javalib/src/main/scala/java/util/Arrays.scala

```diff
@@ -2,167 +2,160 @@ package java.util
+    sortRangeImpl[Int](a, fromIndex, toIndex)
 
-  def fill(a: Array[Byte], value: Byte): Unit =
-    fillImpl(a, value)
+  @noinline def sort(a: Array[Long]): Unit =
+    scala.util.Sorting.stableSort(a)
 
-  def fill(a: Array[Byte], fromIndex: Int, toIndex: Int, value: Byte): Unit =
-    fillImpl(a, fromIndex, toIndex, value)
+  @noinline def sort(a: Array[Long], fromIndex: Int, toIndex: Int): Unit =
+    sortRangeImpl[Long](a, fromIndex, toIndex)
 
-  def fill(a: Array[Char], value: Char): Unit =
-    fillImpl(a, value)
+  @noinline def sort(a: Array[Short]): Unit =
+    scala.util.Sorting.stableSort(a)
 
-  def fill(a: Array[Char], fromIndex: Int, toIndex: Int, value: Char): Unit =
-    fillImpl(a, fromIndex, toIndex, value)
+  @noinline def sort(a: Array[Short], fromIndex: Int, toIndex: Int): Unit =
+    sortRangeImpl[Short](a, fromIndex, toIndex)
 
-  def fill(a: Array[Short], value: Short): Unit =
-    fillImpl(a, value)
+  @noinline def sort(a: Array[Char]): Unit =
+    scala.util.Sorting.stableSort(a)
 
-  def fill(a: Array[Short], fromIndex: Int, toIndex: Int, value: Short): Unit =
-    fillImpl(a, fromIndex, toIndex, value)
+  @noinline def sort(a: Array[Char], fromIndex: Int, toIndex: Int): Unit =
+    sortRangeImpl[Char](a, fromIndex, toIndex)
 
-  def fill(a: Array[Int], value: Int): Unit =
-    fillImpl(a, value)
+  @noinline def sort(a: Array[Byte]): Unit =
+    scala.util.Sorting.stableSort(a)
 
-  def fill(a: Array[Int], fromIndex: Int, toIndex: Int, value: Int): Unit =
-    fillImpl(a, fromIndex, toIndex, value)
+  @noinline def sort(a: Array[Byte], fromIndex: Int, toIndex: Int): Unit =
+    sortRangeImpl[Byte](a, fromIndex, toIndex)
 
-  def fill(a: Array[Long], value: Long): Unit =
-    fillImpl(a, value)
+  @noinline def sort(a: Array[Float]): Unit =
+    scala.util.Sorting.stableSort(a)
 
-  def fill(a: Array[Long], fromIndex: Int, toIndex: Int, value: Long): Unit =
-    fillImpl(a, fromIndex, toIndex, value)
+  @noinline def sort(a: Array[Float], fromIndex: Int, toIndex: Int): Unit =
+    sortRangeImpl[Float](a, fromIndex, toIndex)
 
-  def fill(a: Array[Float], value: Float): Unit =
-    fillImpl(a, value)
+  @noinline def sort(a: Array[Double]): Unit =
+    scala.util.Sorting.stableSort(a)
 
-  def fill(a: Array[Float], fromIndex: Int, toIndex: Int, value: Float): Unit =
-    fillImpl(a, fromIndex, toIndex, value)
+  @noinline def sort(a: Array[Double], fromIndex: Int, toIndex: Int): Unit =
+    sortRangeImpl[Double](a, fromIndex, toIndex)
 
-  def fill(a: Array[Double], value: Double): Unit =
-    fillImpl(a, value)
+  @noinline def sort(a: Array[AnyRef]): Unit =
+    scala.util.Sorting.stableSort(a)
 
-  def fill(a: Array[Double], fromIndex: Int, toIndex: Int, value: Double): Unit =
-    fillImpl(a, fromIndex, toIndex, value)
+  @noinline def sort(a: Array[AnyRef], fromIndex: Int, toIndex: Int): Unit = {
+    implicit val tag = ClassTag[AnyRef](a.getClass.getComponentType)
```

Comment:
There's something wrong here. You shouldn't be able to create a `ClassTag[AnyRef]` for something that's not guaranteed to be `AnyRef`.
I don't think you even need that `ClassTag`. The implicit `ClassTag` materialized for `AnyRef` automatically will do the trick.